app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:meta/meta.dart';
+
+import '../../cocoon_service.dart';
+import '../model/appengine/service_account_info.dart';
+import '../model/luci/buildbucket.dart';
+import 'buildbucket.dart';
+
+/// Class with utilities to interact with LUCI buildbucket to get, trigger
+/// and cancel builds for github repos. It uses [config.luciTryBuilders] to
+/// get the list of available builders.
+class LuciBuildService {
+  LuciBuildService(this.config, this.buildBucketClient, this.serviceAccount);
+
+  BuildBucketClient buildBucketClient;
+  Config config;
+  ServiceAccountInfo serviceAccount;
+  List<Status> failStatusList = <Status>[
```

Comment:
Done.

> Thanks for the refactoring to avoid duplicating LUCI interactions everywhere.
> 
> Are follow up PRs going to replace the duplicated functionalities in luci_status.dart and github_webhook.dart with this service?

Yes, the next PR will remove the duplicated code.